compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/SwingPanel.desktop.kt

```diff
@@ -75,23 +81,19 @@ public fun <T : Component> SwingPanel(
     val componentInfo = remember { ComponentInfo<T>() }
 
     val root = LocalLayerContainer.current
-    val density = LocalDensity.current.density
+    val density = LocalDensity.current
     val focusManager = LocalFocusManager.current
     val focusSwitcher = remember { FocusSwitcher(componentInfo, focusManager) }
 
     Box(
         modifier = modifier.onGloballyPositioned { childCoordinates ->
             val coordinates = childCoordinates.parentCoordinates!!
-            val location = coordinates.localToWindow(Offset.Zero).round()
-            val size = coordinates.size
-            componentInfo.container.setBounds(
-                (location.x / density).toInt(),
-                (location.y / density).toInt(),
-                (size.width / density).toInt(),
-                (size.height / density).toInt()
-            )
+            val bounds = coordinates.boundsInWindow().round(density)
+            componentInfo.container.setBounds(bounds.left, bounds.top, bounds.width, bounds.height)
             componentInfo.container.validate()
             componentInfo.container.repaint()
+        }.drawBehind {
+            drawRect(Color.Transparent, blendMode = BlendMode.Clear) // draw transparent hole
```

Comment:
It still doesn't work on Windows:
<img width="400" alt="image" src="https://github.com/JetBrains/compose-multiplatform-core/assets/5963351/80f993e5-d76a-4b9c-aeb5-8cf0d6d3a498">

It seems we have a reverse order on Windows (on Linux probably too) - showing Swing on top of Compose 🫠

Reply:
Fixed D3D inside skiko: https://github.com/JetBrains/skiko/commit/ed3a1ef6efbfa3607c9bbd2de6e0f1fa33d36724